vegbilder-client/src/hooks/useFetchImagePointsFromOGC.ts

```diff
@@ -1,32 +1,94 @@
 import { useState } from 'react';
 
 import getImagePointsInTilesOverlappingBbox from 'apis/VegbilderOGC/getImagePointsInTilesOverlappingBbox';
-import { IBbox } from 'types';
-import { useSetRecoilState } from 'recoil';
-import { loadedImagePointsFilterState } from 'recoil/selectors';
+import { IBbox, IImagePoint } from 'types';
+import { useRecoilValue, useRecoilState } from 'recoil';
+import { loadedImagePointsFilterState, availableYearsQuery } from 'recoil/selectors';
+import { currentImagePointState } from 'recoil/atoms';
+import { getNearestImagePointInSameDirectionOfImagePoint } from 'utilities/imagePointUtilities';
 
 const useFetchImagePointsFromOGC = () => {
   const [isFetching, setIsFetching] = useState(false);
-  const setLoadedImagePoints = useSetRecoilState(loadedImagePointsFilterState);
+  const [loadedImagePoints, setLoadedImagePoints] = useRecoilState(loadedImagePointsFilterState);
+  const currentImagePoint = useRecoilValue(currentImagePointState);
+  const availableYearsForAllImageTypes = useRecoilValue(availableYearsQuery);
+
+  const imageTypeHasImagesForYear = (imageType: string, year: number) => {
+    return availableYearsForAllImageTypes[imageType].includes(year);
+  }
+
 
   async function fetchImagePointsByYearAndBbox(year: number, bbox: IBbox) {
     if (isFetching) return;
     setIsFetching(true);
-    const { imagePoints, expandedBbox } = await getImagePointsInTilesOverlappingBbox(bbox, year);
-    console.info('Antall bildepunkter returnert fra ogc: ' + imagePoints.length);
-    if (imagePoints && imagePoints.length > 0) {
-      setLoadedImagePoints({
-        imagePoints: imagePoints,
-        bbox: expandedBbox,
-        year: year,
-      });
-      setIsFetching(false);
-      return imagePoints;
+
+
+    let imagePointsAll: IImagePoint[] = [];
+    let imagePointsPlanar: IImagePoint[] = [];
+    let imagePointsPanorama: IImagePoint[] = [];
+    let expandedBbox: any = {};
+
+      let resPlanar: any = {};
+      let res360: any = {};
+
+      if (imageTypeHasImagesForYear('planar', year)) {
+        resPlanar = await getImagePointsInTilesOverlappingBbox(bbox, `vegbilder_1_0:Vegbilder_${year}`);
+      }
+      if (imageTypeHasImagesForYear('panorama', year)) {
+        res360 = await getImagePointsInTilesOverlappingBbox(bbox, `vegbilder_1_0:Vegbilder_360_${year}`);
+      }
```

Comment:
```suggestion
      const resPlanar: any = imageTypeHasImagesForYear('planar', year)
      ? await getImagePointsInTilesOverlappingBbox(bbox, `vegbilder_1_0:Vegbilder_${year}`)
      : {};
    const res360: any = imageTypeHasImagesForYear('panorama', year)
      ? await getImagePointsInTilesOverlappingBbox(bbox, `vegbilder_1_0:Vegbilder_360_${year}`)
      : {};
```
Vil sterkt anbefale å bruke const, og initialisere variablar til "riktig" verdi med ein gong i staden for å sette dei til ein verdi først og så mutere dei etterpå. Kan seie meir om det muntleg.

Reply:
Trur heller ikkje det er så mykje arbeid å få fjerna den anyen der, men gidd ikkje å skrive det ut. Kan ta det muntleg om du vil.